packages/mux-player-react/src/lazy.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+
+import ConditionalSuspense from './ConditionalSuspense';
+import useIsBrowser from './useIsBrowser';
+import useIsIntersecting from './useIsIntersecting';
+
+import type { MuxPlayerProps, MuxPlayerRefAttributes } from './index';
+
+const MuxPlayerIndex = React.lazy(() => import('./index'));
+
+interface FallbackProps extends MuxPlayerProps {
+  onIntersection?: () => void;
+}
+const Fallback = (props: FallbackProps) => {
+  const { style, className, onIntersection, placeholder } = props;
+
+  const intersectionRef = React.useRef<HTMLElement>(null);
+  const isIntersecting = useIsIntersecting(intersectionRef);
+
+  useEffect(() => {
+    if (isIntersecting && onIntersection) {
+      onIntersection();
+    }
+  }, [isIntersecting, onIntersection]);
+
+  return (
+    <>
+      {/* 
+        Why do we have a mux-player element before the mux-player bundle is even loaded?
+        Before the bundle is loaded, this mux-player element just acts like a div. 
+        However, by calling this placeholder "mux-player", 
+        it now gets the same CSS applied to it that the eventual "real" mux-player element will.
+      */}
+      {/* TODO: can we add mux-player to JSX.IntrinsicElements */}
```

Comment:
Would anybody be interested in trying this out for me? For the life of me I couldn't figure it out.

Reply:
@cjpillsbury